rust_src/src/character.rs

```diff
@@ -92,4 +92,13 @@ pub fn multibyte_char_to_unibyte(ch: Codepoint) -> EmacsInt {
     }
 }
 
+/// Resolve modifiers in the character CHAR.
+/// The value is a character with modifiers resolved into the character
+/// code.  Unresolved modifiers are kept in the value.
+/// usage: (char-resolve-modifiers CHAR)
+#[lisp_fn]
+pub fn char_resolve_modifiers(character: LispObject) -> EmacsInt {
```

Comment:
```suggestion
pub fn char_resolve_modifiers(character: CodePoint) -> EmacsInt {
```

If we did this, we would not need to perform the transformation inside the function. What do you think?

Reply:
If both the argument and return type was changed, the method could simply be called directly, i.e. replacing the function body with `character.resolve_modifier_mask()`

Reply:
If we change the argument to a `Codepoint`, then we can no longer handle negative numbers being passed in, which emacs currently accepts. Is this desired? If so, I'll do as @agraven suggests.

Reply:
If I understood the discussion in gitter correctly, we're currently imitating emacs 26's behaviour of not accepting negative number, and handling negative numbers was introduced in emacs 27. If you want to test this you can temporarily rename your port to something new, and keep the old function alongside it. You can then compare their outputs.

Reply:
@agraven the question is, when we catch up and need to support negative values can we continue to use `Codepoint` or will we need a new type? If so, perhaps we should consider migrating to that type now even if we do not yet support negative values?

Reply:
@shaleh Can this PR be merged as is (assuming no other changes need to be made)? We should track the potential `Codepoint` change as a separate issue IMO.

Reply:
@shaleh Any updates as to the status of this PR? I know you've been busy!